src/assert.sh

```diff
@@ -318,21 +318,23 @@ function assert_greater_or_equal_than() {
 
 function assert_line_count() {
   local expected="$1"
-  local actual="$2"
+  local input_str="$2"
   local label="${3:-$(helper::normalize_test_function_name "${FUNCNAME[1]}")}"
 
-  if [ -z "$actual" ]; then
-    local actual_line_count=0
+  if [ -z "$input_str" ]; then
+    local actual=0
   else
-    local actual_line_count
-    actual_line_count=$(echo "$actual" | wc -l | tr -d '[:blank:]')
+    local actual
+    actual=$(echo "$input_str" | wc -l | tr -d '[:blank:]')
+    additional_new_lines=$(grep -o '\\n' <<< "$input_str" | wc -l | tr -d '[:blank:]')
```

Comment:
I don't get why/how bash represents the newline in 2 different characters.. but the tests seem to show it works.

I don't know why 😅

Reply:
Same here... I just got to this solution driven by the tests, I don't understand why this works... so it's bash... 🤣 
> And here, yet, another proof that TDD works 😝